### ISSUE TITLE
Decorator based Controller doesn't comply Liskov substitution principle implemented by Hibernate

### ISSUE THREAD

**Issue by Leroy14** [phoenixnap/springmvc-raml-plugin#173]
When the springmvc-raml-plugin is initialized with the rule: 'com.phoenixnap.oss.ramlapisync.generation.rules.Spring4ControllerDecoratorRule' the MVC Rest Controller is implemented by:
- An interface Controller, 
- A Decorator that is composed by a Controller. 
Note: The Decorator implements the interface Controller.

Because the Decorator methods are annotated adding restrictions on their arguments and the overridden methods of the interface are not annotated, the Decorator is not compliant with Liskov substitution principle.

The issue is that from the version >= 1.5.x of springboot, Hibernate validates the Bean against this principle and generates the error ‘Hibernate Validator - HV000151’ for our Decorator that prevents the service to process requests.

A solution would be to remove the overriding link between the Decorator and the interface Controller. Not sure this link is really relevant. 

_What do you think about this issue?_

Note: There is a workaround that consists to disable the Hibernate validation but it is not really recommended.

**Comment by stojsavljevic** [comment]
Hi,

thanks for reaching us.

Can you please add some more details to the issue - `raml` part and Maven configuration so I can easily reproduce it?

Thanks.

**Comment by Leroy14** [comment]
Here is a project that may help you to reproduce this issue.
[raml-173.zip](https://github.com/phoenixnap/springmvc-raml-plugin/files/1258447/raml-173.zip)
Procedure:
* mvn clean install
* mvn spring-boot:run
* send a request GET to http://localhost:8080/resources/a_resource
* confirm that the following error is displayed on the App console
```
2017-08-28 23:29:21.205 ERROR 13476 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is javax.validation.ConstraintDeclarationException: HV000151: A method overriding another method must not alter the parameter constraint configuration, but method public org.springframework.http.ResponseEntity org.test.ResourceAResourceControllerDecorator.getAResource(java.lang.String) changes the configuration of public abstract org.springframework.http.ResponseEntity org.test.ResourceAResourceController.getAResource(java.lang.String).] with root cause
```
In the source code, the ControllerDecorator implements Controller. But the decorator adds a constraint to one of its argument that make the decorator more strict than the interface.
If you remove the link ControllerDecorator ---|> Controller, the Get works.
Note: you need to use http headers with contraint to generate the right method
Hope this will help you to find the right solution

**Comment by stojsavljevic** [comment]
The (most easy) solution will be to add annotations also on interface.
It will be fixed as part of #175
